pkg/remote/client.go

```diff
@@ -0,0 +1,84 @@
+package remote
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/golang/snappy"
+	prompb "go.buf.build/grpc/go/prometheus/prometheus"
+)
+
+type HTTPConfig struct {
+	UserAgent string
+	Timeout   time.Duration
+	TLSConfig tls.Config
+	BasicAuth BasicAuth
+	Headers   http.Header
+}
+
+type BasicAuth struct {
+	User, Password string
+}
+
+type WriteClient struct {
+	hc  *http.Client
+	url *url.URL
+	cfg *HTTPConfig
+}
+
+func NewWriteClient(endpoint string, cfg *HTTPConfig) (*WriteClient, error) {
+	if cfg == nil {
+		cfg = &HTTPConfig{}
+	}
+	u, err := url.Parse(endpoint)
+	if err != nil {
+		return nil, err
+	}
+	return &WriteClient{
+		hc: &http.Client{
+			Timeout: cfg.Timeout,
+		},
+		url: u,
+		cfg: cfg,
+	}, nil
+}
+
+// Store sends a batch of samples to the HTTP endpoint,
+// the request is the proto marshalled and encoded.
+func (c *WriteClient) Store(ctx context.Context, series []*prompb.TimeSeries) error {
+	b, err := proto.Marshal(&prompb.WriteRequest{
+		Timeseries: series,
+	})
+	if err != nil {
+		return fmt.Errorf("encoding series as protobuf write request failed: %w", err)
+	}
+	b = snappy.Encode(nil, b) // TODO: this call can panic
+	req, err := http.NewRequest(http.MethodPost, c.url.String(), bytes.NewReader(b))
+	if err != nil {
+		return fmt.Errorf("create new HTTP request failed: %w", err)
+	}
+
+	req.Header.Add("Content-Encoding", "snappy")
+	req.Header.Set("Content-Type", "application/x-protobuf")
+	req.Header.Set("User-Agent", "k6-prometheus-rw-output")
+	req.Header.Set("X-Prometheus-Remote-Write-Version", "0.1.0")
```

Comment:
This could easily be missed to be upgraded with the release, so maybe we need some variable Version as we have in the other projects. On the other side if we merge the extension to the k6 there should be a k6 version.

So up to you if we should change something here :)

Reply:
Same as for the test, it is defined from the protocol